File: python_open/gaussianblur.py
import cv2

folder = "/home/test/Desktop/opencv/opencv/cppOpen/build/data/"

# 첫 번째 이미지 읽기
src1 = cv2.imread(folder + "img7.bmp", cv2.IMREAD_GRAYSCALE)
if src1 is None:
    print("Error: Could not load image!")
    exit(-1)

# 블러 효과 적용 (가우시안 블러 사용)
kernel_size = 5  # 커널 크기 (홀수여야 함)
dst = cv2.GaussianBlur(src1, (kernel_size, kernel_size), 0)  # 시그마 X: 0은 자동 계산

# 원본 이미지와 블러 이미지 표시
cv2.namedWindow("Original Image", cv2.WINDOW_AUTOSIZE)
cv2.imshow("Original Image", src1)

cv2.namedWindow("Blurred Image", cv2.WINDOW_AUTOSIZE)
cv2.imshow("Blurred Image", dst)

cv2.waitKey(0)  # 키 입력 대기
cv2.destroyAllWindows()

File: cppOpen/gaussianblur.cpp
#include "opencv2/freetype.hpp"
#include "opencv2/opencv.hpp"
#include <iostream>

using namespace cv;
using namespace std;

String folder = "/home/test/Desktop/opencv/opencv/cppOpen/build/data/";

int main() {
    // 첫 번째 이미지 읽기
    Mat src1 = imread(folder + "img7.bmp", IMREAD_GRAYSCALE);
    if (src1.empty()) {
        cout << "이미지를 열 수 없거나 찾을 수 없습니다!" << endl;
        return -1;
    }

    // 블러 효과 적용 (가우시안 블러 사용)
    Mat dst;
    int kernelSize = 5; // 커널 크기 (홀수여야 함)
    GaussianBlur(src1, dst, Size(kernelSize, kernelSize), 0); // 시그마 X: 0은 자동 계산

    // 원본 이미지와 블러 이미지 표시
    namedWindow("Original Image", WINDOW_AUTOSIZE);
    imshow("Original Image", src1);

    namedWindow("Blurred Image", WINDOW_AUTOSIZE);
    imshow("Blurred Image", dst);

    waitKey(0); // 키 입력 대기
    return 0;
}
